Example-Swift/ViewController.swift

```diff
@@ -9,102 +9,69 @@
+        present(photosViewController, animated: true, completion: nil)
     }
-    
-    func updateImagesOnPhotosViewController(photosViewController: NYTPhotosViewController, afterDelayWithPhotos: [ExamplePhoto]) {
-        
-        let delayTime = dispatch_time(DISPATCH_TIME_NOW, 5 * Int64(NSEC_PER_SEC))
-        
-        dispatch_after(delayTime, dispatch_get_main_queue()) {
-            for photo in self.photos {
-                if photo.image == nil {
-                    photo.image = UIImage(named: PrimaryImageName)
-                    photosViewController.updateImageForPhoto(photo)
-                }
-            }
-        }
-    }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        let buttonImage = UIImage(named: PrimaryImageName)
-        imageButton?.setBackgroundImage(buttonImage, forState: .Normal)
+
+        let buttonImage = UIImage(named: ReferencePhotoName)
+        imageButton?.setBackgroundImage(buttonImage, for: UIControlState())
     }
-    
-    // MARK: - NYTPhotosViewControllerDelegate
-    
-    func photosViewController(photosViewController: NYTPhotosViewController, handleActionButtonTappedForPhoto photo: NYTPhoto) -> Bool {
+}
 
-        if UIDevice.currentDevice().userInterfaceIdiom == .Pad {
-            
+// MARK: NYTPhotosViewControllerDelegate
+
+extension ViewController: NYTPhotosViewControllerDelegate {
+    
+    func photosViewController(_ photosViewController: NYTPhotosViewController, handleActionButtonTappedFor photo: NYTPhoto) -> Bool {
+        if UIDevice.current.userInterfaceIdiom == .pad {
             guard let photoImage = photo.image else { return false }
             
             let shareActivityViewController = UIActivityViewController(activityItems: [photoImage], applicationActivities: nil)
-            
-            shareActivityViewController.completionWithItemsHandler = {(activityType: String?, completed: Bool, items: [AnyObject]?, error: NSError?) in
+            shareActivityViewController.completionWithItemsHandler = {(activityType: UIActivityType?, completed: Bool, items: [Any]?, error: Error?) in
                 if completed {
-                    photosViewController.delegate?.photosViewController!(photosViewController, actionCompletedWithActivityType: activityType!)
+                    photosViewController.delegate?.photosViewController!(photosViewController, actionCompletedWithActivityType: activityType?.rawValue)
                 }
             }
 
             shareActivityViewController.popoverPresentationController?.barButtonItem = photosViewController.rightBarButtonItem
-            photosViewController.presentViewController(shareActivityViewController, animated: true, completion: nil)
+            photosViewController.present(shareActivityViewController, animated: true, completion: nil)
 
             return true
         }
         
         return false
     }
```

Comment:
Done in https://github.com/NYTimes/NYTPhotoViewer/pull/223/commits/141e05bb820ccb27daf361502c2ab4a50751dfda